src/components/capture/creation/index.tsx

```diff
@@ -58,48 +70,40 @@ function NewCaptureModal() {
 
     // Form Event Handlers
     const handlers = {
-        close: () => {
-            navigate('..'); //This is assuming this is a child of the /captures route.
-        },
-
-        delete: () => {
-            console.log('Delete? You sure?');
-        },
-
-        save: (event: MouseEvent<HTMLElement>) => {
+        addToChangeSet: (event: MouseEvent<HTMLElement>) => {
             event.preventDefault();
-            let catalogVal = '';
 
-            if (editorRef.current) {
-                catalogVal = editorRef.current.getValue();
-            }
+            const namespace: string = details.data.name;
 
-            setFormSubmitting(true);
+            // TODO: Get user identifier from authentication-related application state.
+            const capture: Entity = {
+                metadata: {
+                    changeType: 'New Entity',
+                    entityType: 'Capture',
+                    name: namespace.substring(
+                        namespace.lastIndexOf('/') + 1,
+                        namespace.length
+                    ),
+                    namespace,
+                    user: 'temp@gmail.com',
+                },
+                schema: schemaFromEditor || catalogResponse,
+            };
 
-            // Create blob link to download
-            const url = window.URL.createObjectURL(
-                new Blob([catalogVal], {
-                    type: 'text/plain',
-                })
-            );
+            addCaptureToChangeSet(namespace, capture);
+            removeSchema();
 
-            // Make download link
-            const link = document.createElement('a');
-            link.href = url;
-            link.setAttribute('download', `${details.data.name}.flow.yaml`);
+            setFormSubmitting(true);
 
-            // Append to html link element page
-            document.body.appendChild(link);
+            window.setTimeout(() => handlers.close(), 0);
```

Comment:
Why the timeout?

Reply:
**PENDING CORRECTION.** Oh, I laughed when I saw this line of code after opening the PR. There should be no timeout. It was a C&P oversight uncaught while programming due to the events of the day. As aforementioned, this PR was meant to be a draft to review and correct any mistakes before anyone else took a critical look, but I forgot to press that button yesterday evening.